connect/models/marketplace.py

```diff
@@ -57,49 +76,71 @@ class AgreementSchema(BaseSchema):
-    owner = fields.Nested(CompanySchema, only=('id', 'name'))
-    stats = fields.Dict()
-    active = fields.Bool()
+    owner = fields.Nested(CompanySchema)
+    stats = fields.Nested(AgreementStatsSchema, allow_none=True)
+    author = fields.Nested(CompanySchema, allow_none=True)
     version = fields.Int()
+    active = fields.Bool()
     link = fields.Str()
     version_created = fields.DateTime()
     version_contracts = fields.Int()
+    agreements = fields.Nested('AgreementSchema', many=True)
+    parent = fields.Nested('AgreementSchema', only=('id', 'name'), allow_none=True)
+    marketplace = fields.Nested(MarketplaceSchema, only=('id', 'name'), allow_none=True)
 
     @post_load
     def make_object(self, data):
         return Agreement(**data)
 
 
+class Activation(BaseModel):
+    link = None  # type: Optional[str]
+    message = None  # type: str
+    date = None  # type: Optional[str]
+
+
+class ActivationSchema(BaseSchema):
+    link = fields.Str(allow_none=True)
+    message = fields.Str()
+    date = fields.DateTime(allow_none=True)
+
+    @post_load
+    def make_object(self, data):
+        return Activation(**data)
+
+
 class Contract(BaseModel):
     name = None  # type: str
-    status = None  # type: str
     version = None  # type: int
     type = None  # type: str
+    status = None  # type: str
     agreement = None  # type: Agreement
-    marketplace = None  # type: Marketplace
-    owner = None  # type: Company
+    marketplace = None  # type: Optional[Marketplace]
+    owner = None  # type: Optional[Company]
+    creator = None  # type: Company
     created = None  # type: str
     updated = None  # type: str
-    enrolled = None  # type: str
+    enrolled = None  # type: Optional[str]
     version_created = None  # type: str
-    activation = None  # type: dict
-    signee = None  # type: Company
+    activation = None  # type: Activation
+    signee = None  # type: Optional[Company]
 
 
 class ContractSchema(BaseSchema):
     name = fields.Str()
-    status = fields.Str()
     version = fields.Int()
     type = fields.Str()
+    status = fields.Str()
     agreement = fields.Nested(AgreementSchema, only=('id', 'name'))
-    marketplace = fields.Nested(MarketplaceSchema, only=('id', 'name'))
-    owner = fields.Nested(CompanySchema, only=('id', 'name'))
+    marketplace = fields.Nested(MarketplaceSchema, only=('id', 'name'), allow_none=True)
+    owner = fields.Nested(CompanySchema, only=('id', 'name'), allow_none=True)
+    creator = fields.Nested(CompanySchema, only=('id', 'name'))
     created = fields.DateTime()
     updated = fields.DateTime()
-    enrolled = fields.Str()
+    enrolled = fields.Str(allow_none=True)
     version_created = fields.DateTime()
-    activation = fields.Dict()
-    signee = fields.Nested(CompanySchema, only=('id', 'name'))
+    activation = fields.Nested(ActivationSchema)
+    signee = fields.Nested(CompanySchema, only=('id', 'name'), allow_none=True)
```

Comment:
Multiple such places, probably:
signee = OptionalReference(CompanySchema)
or 
creator = Reference(UserSchema)

Reply:
Once we decide how to do it in the other conversation, I'll fix it on all places.